sdk/helper/storagepacker/ids2keys.go

```diff
@@ -0,0 +1,99 @@
+package storagepacker
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/vault/sdk/helper/cryptoutil"
+	"math"
+	"sort"
+	"strings"
+)
+
+type itemRequest struct {
```

Comment:
Why call it itemRequest? What does the request refer to here?

Reply:
GetItems() takes a list of IDs. I want the returned items to respect the original order. So, this struct is a wrapper that maintains the initial request ordering for the items, as well as caching the key for each item ID.

Perhaps we could just use Item, but make `key` a non-exported member of the struct.

Or, it could be named something more verbose like wrappedRequest or wrappedItem.

Reply:
Merged itemRequest and Item so we only use Item any more.